crypto/fipsmodule/kem/kem.c

```diff
@@ -306,3 +308,45 @@ int KEM_KEY_set_raw_key(KEM_KEY *key, const uint8_t *in_public,
 
   return 1;
 }
+
+int KEM_KEY_set_raw_keypair_from_seed(KEM_KEY *key, const CBS *seed) {
+  if (key == NULL || seed == NULL || key->kem == NULL) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_INVALID_PARAMETERS);
+    return 0;
+  }
+
+  // Validate seed length - all ML-KEM variants use 64-byte seeds
+  if (CBS_len(seed) != key->kem->keygen_seed_len) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_INVALID_BUFFER_SIZE);
+    return 0;
+  }
+
+  // Allocate temporary buffers for key generation
+  uint8_t *public_key = OPENSSL_malloc(key->kem->public_key_len);
+  uint8_t *secret_key = OPENSSL_malloc(key->kem->secret_key_len);
+  if (public_key == NULL || secret_key == NULL) {
+    OPENSSL_free(public_key);
+    OPENSSL_free(secret_key);
+    OPENSSL_PUT_ERROR(EVP, ERR_R_MALLOC_FAILURE);
```

Comment:
NP: Not necessary. This is set by `OPENSSL_malloc` on a failure.

Reply:
Thanks, removed in OPENSSL_PUT_ERROR(CRYPTO, ERR_R_OVERFLOW);